modules/console/pages/ui/programmable-push-filters.adoc

```diff
@@ -2,58 +2,172 @@
-the JavaScript function. If your record is presented as a JSON object in the UI, you can also access it
-like a JavaScript object in your filter code.
+. Navigate to the topic's *Messages* page.
+. Click *Add filter > JavaScript Filter*.
+. Define your JavaScript filtering logic in the provided input area.
 
-Suppose you have a series of Avro, JSON, or Protobuf encoded record values that deserialize to JSON objects like this:
+image::ROOT:console:js-filter.png[alt="JavaScript filter in {ui}"]
 
-[,json]
+== Resource usage and performance
+JavaScript filters are executed on the backend, consuming CPU and network resources. The performance of your filter depends on the complexity of your JavaScript code and the volume of data being processed.
+Complex JavaScript logic or large data volumes may increase CPU load and network usage.
+
+== Available JavaScript properties
+
+{ui} injects these properties into your JavaScript context:
+
+[cols="1a,2a,1a"]
+|===
+| Property         | Description                                      | Type
+
+| `headers`        | Record headers as key-value pairs (ArrayBuffers) | Object
+| `key`            | Decoded record key                               | String
+| `keySchemaID`    | Schema Registry ID for key (if present)          | Number
+| `partitionId`    | Partition ID of the record                       | Number
+| `offset`         | Record offset within partition                   | Number
+| `timestamp`      | Timestamp as JavaScript Date object              | Date
+| `value`          | Decoded record value                             | Object/String
+| `valueSchemaID`  | Schema Registry ID for value (if present)        | Number
+|===
+
+NOTE: Values, keys, and headers are deserialized before being injected into your script.
+
+== JavaScript filter examples
+
+=== Filter by header value
+
+*Scenario:* Records tagged with headers specifying customer plan type.
+
+.Sample header data (string value)
+[source,json]
+----
+headers: {
+  "plan_type": "premium"
+}
+----
+
+.JavaScript filter
+[source,javascript]
+----
+let headerValue = headers["plan_type"];
+if (headerValue) {
+  let stringValue = String.fromCharCode(...new Uint8Array(headerValue));
+  return stringValue === "premium";
+}
+return false;
+----
+
+*Scenario:* Records include a header with JSON-encoded customer metadata.
+
+.Sample header data (JSON value)
+[source,json]
+----
+headers: {
+"customer": "{"orgID":"123-abc","name":"ACME Inc."}"
+}
+----
+
+.JavaScript filter
+[source,javascript]
+----
+let headerValue = headers["customer"];
+if (headerValue) {
+  let stringValue = String.fromCharCode(headerValue);
+  let valueObj = JSON.parse(stringValue);
+  return valueObj["orgID"] === "123-abc";
+}
+return false;
+----
+
+=== Filter by timestamp
+
+*Scenario:* Retrieve records from a promotional event (e.g., Black Friday on November 24).
```

Comment:
Some people have a negative connotation of Black Friday, and some people may have never heard of it. Maybe just use "sale" or something instead.